ibis/common/grounds.py

```diff
@@ -270,15 +270,15 @@ def map(self, fn, filter=None):
         results = {}
         for node in Graph.from_bfs(self, filter=filter).toposort():
             kwargs = recursive_get(node.__getstate__(), results)
-            results[node] = fn(node, **kwargs)
+            results[node] = fn(node, **kwargs, cache=results)
```

Comment:
Rather pass it as positional so we don't interfere with kwargs?

Reply:
Sounds good.

Reply:
Done!